test/libsolidity/semanticTests/array/copying/copy_dynamically_sized_storage_array.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
```

Comment:
The fact that the array contains function pointers is the key point here so it should be included in the name. Looking at naming in this test dir, I'd call your new test cases something like
- `copy_dynamically_sized_storage_array.sol` -> `array_of_function_external_storage_to_storage_dynamic.sol`
- `copy_storage_arrays_of_function_types.sol` -> `array_of_function_external_storage_to_storage_dynamic_different_mutability.sol`

The existing `copy_function_storage_array.sol` should also really have `internal` somewhere in its name.